Add how to Post/Get/List orders (#2)
spacefill/rest-api-tutorial: README.md

This tutorial is based on [`curl`](https://en.wikipedia.org/wiki/CURL) command line tool.

- [Environments](#environments)
- [Getting started](#getting-started)
- [CRUD list](#crud-list)
- [How can a client get orders.status update (events) information?](#how-can-a-client-get-ordersstatus-update-events-information)
  - [Client pull method](#client-pull-method)
  - [Server push method with webhook](#server-push-method-with-webhook)

## Environments

> Two environments are availables:

Reviewer: ```suggestion
Two environments are available:
```

> - The `production` at `https://api.spacefill.fr`

Reviewer: ```suggestion
- The `production` one at `https://api.spacefill.fr`
```

> - The `sandbox` at `https://api.sandbox.spacefill.fr` that can be used to try API

Reviewer: ```suggestion
- The `sandbox` one at `https://api.sandbox.spacefill.fr` that can be used to try the API
```

> To access to those environments, you need:

Reviewer: ```suggestion
To access those environments, you need:
```

> - An security API Token

Reviewer: ```suggestion
- An API security Token
```
ou
```suggestion
- A secure API Token
```

> - The warehouse ids where you want store your merchandises

Reviewer: ```suggestion
- The warehouses where you want store your merchandises ids
```
ou
```suggestion
- The ids for the warehouses where you want store your merchandises
```
Eventuellement préciser "SpaceFill ids" ?

> Ask directly this informations to <gustave@spacefill.fr>.

Reviewer: ```suggestion
Ask directly this information to <gustave@spacefill.fr>.
```

SpaceFill API documentation reference is available at (powered by [Swagger](https://swagger.io/)):

- https://api.spacefill.fr/docs
- https://api.sandbox.spacefill.fr/docs

## Getting started

> Before executing any requests you must setup your token, it must be used in header as `x_token`.

Reviewer: suggestion de précision

```suggestion
Before executing any requests you must setup your token, it must be used as `x_token` header value.
```
Pas forcément nécessaire

  "comment": "Additional comment",
  "planned_execution_datetime_range": {
    "datetime_from": "2021-09-28T15:12:41.538Z",
    "datetime_to": "2021-09-28T15:12:41.538Z"
  },
  "carrier_name": "Carrier X",
  "carrier_phone_number": "+33610101010",
  "transport_management_owner": "PROVIDER",
  "entry_expeditor": "Expeditor X",
  "entry_expeditor_address_line1": "123 Boulevard X",
  "entry_expeditor_address_line2": "",
  "entry_expeditor_address_line3": "",
  "entry_expeditor_address_zip": "75020",
  "entry_expeditor_address_city": "Paris",
  "entry_expeditor_address_country": "France",
  "entry_expeditor_planned_datetime_range": {
    "datetime_from": "2021-09-28T15:12:41.538Z",
    "datetime_to": "2021-09-28T15:12:41.538Z"
  }
  "order_items": [
    {
      "master_item_id": "13acc10a-a6ab-4099-b600-fb33fa6c0001",
      "batch_id": "e4f77256-304a-4911-a1eb-9f6d47ca0001",
      "item_packaging_type": "PALLET",
      "expected_quantity": 1
    },
    {
      "master_item_id": "13acc10a-a6ab-4099-b600-fb33fa6c0002",
      "batch_id": "e4f77256-304a-4911-a1eb-9f6d47ca0005",
      "item_packaging_type": "PALLET",
      "expected_quantity": 1
    }
  ]
}'
```

> Or an exit `order` with those items:

Reviewer: est-ce qu'il y a besoin d'aller autant dans le détail sachant qu'on a aussi une doc dédiée à orders ?  Entry ne suffirait pas ?

Author: @Claire-Couppie The aim is to have an example from creation to complete order, with entry and exit

  "planned_execution_datetime_range": {
    "datetime_from": "2021-09-28T15:12:41.538Z",
    "datetime_to": "2021-09-28T15:12:41.538Z"
  },
  "carrier_name": "Carrier X",
  "carrier_phone_number": "+33610101010",
  "transport_management_owner": "PROVIDER",
  "exit_final_recipient": "Recipient X",
  "exit_final_recipient_address_line1": "123 Boulevard X",
  "exit_final_recipient_address_line2": "",
  "exit_final_recipient_address_line3": "",
  "exit_final_recipient_address_zip": "75020",
  "exit_final_recipient_address_city": "Paris",
  "exit_final_recipient_address_country": "France",
  "exit_final_recipient_planned_datetime_range": {
    "datetime_from": "2021-09-28T15:12:41.538Z",
    "datetime_to": "2021-09-28T15:12:41.538Z"
  }
  "order_items": [
    {
      "master_item_id": "13acc10a-a6ab-4099-b600-fb33fa6c0001",
      "batch_id": "e4f77256-304a-4911-a1eb-9f6d47ca0001",
      "item_packaging_type": "PALLET",
      "expected_quantity": 1
    },
    {
      "master_item_id": "13acc10a-a6ab-4099-b600-fb33fa6c0002",
      "batch_id": "e4f77256-304a-4911-a1eb-9f6d47ca0005",
      "item_packaging_type": "PALLET",
      "expected_quantity": 1
    }
  ]
}'
```

> You can now view your datas on Spacefill app at `https://app.spacefill.fr`:

Reviewer: Ce n'est pas tout à fait vrai s'ils utilisent la sandbox comme dans l'exemple, si ?

+ data toujours singulier
```suggestion
You can now view your data on Spacefill app at `https://app.spacefill.fr`:
```

- To view your orders go to [`https://app.spacefill.fr/logistic-management/orders/`](https://app.spacefill.fr/logistic-management/orders/)
  [<img src='spacefill-app-orders-view.png' width='600' />](./spacefill-app-orders-view.png)
- To view your master_items go to [`https://app.spacefill.fr/logistic-management/inventory/items/`](https://app.spacefill.fr/logistic-management/inventory/items/)
  [<img src='spacefill-app-master-items-view.png' width='600' />](./spacefill-app-master-items-view.png)

## CRUD list

> You can have more informations about CRUD on each resources using following links:

Reviewer: ```suggestion
You can get more information about CRUD on each resource using the following links:
```

}'
```

After these `master-items` creation, we can create an entry `order` with those items:

```sh
$ curl -sLX 'POST' \
  'https://api.sandbox.spacefill.fr/v1/logistic_management/orders/entry' \
  -H 'accept: application/json' \
  -H 'x_token: secret' \
  -H 'Content-Type: application/json' \
  -d '{
  "order": {
    "shipper_order_reference": "REF_01",
    "warehouse_id": "d8bdc728-242b-4039-99a3-0aa239650011",
    "comment": "Additional comment",
    "planned_execution_datetime_range": {
      "datetime_from": "2021-09-28T15:12:41.538Z",
      "datetime_to": "2021-09-28T15:12:41.538Z"
    },
    "carrier_name": "Carrier X",
    "carrier_phone_number": "+33610101010",
    "transport_management_owner": "PROVIDER",
    "entry_expeditor": "Expeditor X",
    "entry_expeditor_address_line1": "123 Boulevard X",
    "entry_expeditor_address_line2": "",
    "entry_expeditor_address_line3": "",
    "entry_expeditor_address_zip": "75020",
    "entry_expeditor_address_city": "Paris",
    "entry_expeditor_address_country": "France",
    "entry_expeditor_planned_datetime_range": {
      "datetime_from": "2021-09-28T15:12:41.538Z",
      "datetime_to": "2021-09-28T15:12:41.538Z"
    }
  },

>   "order_items": [

Reviewer 2: @Its-Alex **issue:** see https://gitlab.spacefill.fr/spacefill/spacefill/-/issues/6595
